routes/proxy.js

```diff
@@ -0,0 +1,67 @@
+"use strict";
+
+const express = require('express');
+var cookieParser = require('cookie-parser');
+const secretKey = require('../util/secret_key');
+const session = require('express-session');
+const MySQLStore = require('express-mysql-session')(session);
+const db = require('../util/db');
+const { BadRequestError } = require('../util/errors');
+const EngineManager = require('../almond/enginemanagerclient');
+const user = require('../util/user');
+
+const router = express.Router();
+
+const sessionStore = new MySQLStore({
+    expiration: 86400000 // 1 day, in ms
+}, db.getPool());
+
+router.use(cookieParser(secretKey.getSecretKey()));
+router.use(session({ resave: false,
+                        saveUninitialized: false,
+                        store:sessionStore,
+                        secret: secretKey.getSecretKey()}));
```

Comment:
You don't need to setup the session, the server already has the session set up code.